addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/http/ComparableResponse.java

```diff
@@ -468,10 +473,19 @@ private Map<String, Integer> getAllPaths() {
                     }
                 }
                 if (contentType.contains(CONTENT_TYPE_JSON)) {
-                    JSONObject jsonBody = JSONObject.fromObject(body);
-                    for (String childPath : getJsonElementPaths(jsonBody)) {
-                        allPaths.merge(childPath, 1, Integer::sum);
-                        numPaths++;
+                    try {
+                        JSONObject jsonBody = JSONObject.fromObject(body);
+                        for (String childPath : getJsonElementPaths(jsonBody)) {
+                            allPaths.merge(childPath, 1, Integer::sum);
+                            numPaths++;
+                        }
+                    } catch (JSONException je) {
+                        LOGGER.debug(je, je);
+                        JSONArray json = JSONArray.fromObject(body);
```

Comment:
It should handle all types not just arrays and objects (or at the very least not fail again.)

Reply:
I can refactor this to use instanceof instead of the exception I guess. But do you have an example of JSON that will still fail? I'm happy to handle further, I just couldn't think of another case.

Reply:
It will fail with booleans, numbers, and strings. If you are asking for real world examples I'd have to search them (I have encountered cases with those types not just arrays or objects though).

Reply:
Yes, the latest line of examples.

Reply:
I've added a test for this, unless I'm still not understanding then it does currently work.

Reply:
The tests are not correct, the `cloneAll()` only works if the (response) header is non-empty.

Reply:
The content type response header is added before the clone.

Reply:
Adding the header does not make it non-empty, that's based on the status line.

Reply:
Oh okay.

Reply:
Sorted.

Reply:
Use `msg.getResponseHeader().setMessage("HTTP/1.1 200 OK");` instead (needs to be done before adding the header or other changes).